src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/auditing/AnalysisAuditing.java

```diff
@@ -0,0 +1,117 @@
+package ca.corefacility.bioinformatics.irida.ria.web.analysis.auditing;
+
+/*
+ * This class is used for auditing analysis submissions using
+ * AuditReader and EntityManagerFactory.
+ */
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import org.hibernate.envers.AuditReader;
+import org.hibernate.envers.AuditReaderFactory;
+import org.hibernate.envers.query.AuditEntity;
+
+import ca.corefacility.bioinformatics.irida.model.enums.AnalysisState;
+import ca.corefacility.bioinformatics.irida.model.workflow.submission.AnalysisSubmission;
+import ca.corefacility.bioinformatics.irida.ria.web.utilities.DateUtilities;
+
+public class AnalysisAuditing {
+	private EntityManagerFactory entityManagerFactory;
+
+	public AnalysisAuditing(EntityManagerFactory entityManagerFactory) {
+		this.entityManagerFactory=entityManagerFactory;
+	}
+
+	/**
+	 * Gets the running time of an analysis
+	 *
+	 * @param submission {@link AnalysisSubmission} The submission {@link AnalysisSubmission}
+	 * @return {@link Long} Running time of the analysis
+	 */
+	public Long getAnalysisRunningTime(AnalysisSubmission submission) {
+		EntityManager entityManager = entityManagerFactory.createEntityManager();
+		AuditReader audit = AuditReaderFactory.get(entityManager);
```

Comment:
Rather than deal with all this `AuditReader` stuff you should be able to get everything from the `AnalysisSubmissionRepository`.  It extends `RevisionRepository` which should have the methods you need to get the history you're grabbing below.  See docs at https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/history/RevisionRepository.html?is-external=true

Reply:
Thanks Tom! Updated in 6766c8ecf35d0d185900b8e1958aca1fa9d2d409